In [34]:
import numpy as np

# Домашнее задание к уроку 6


## Задание 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [35]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832]) 

In [36]:
# вычисление ковариации с помощью элементарных действий
cov_el = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov_el

9157.839999999997

In [37]:
# вычисление ковариации с помощью функции cov
cov_np =np.cov(zp, ks, ddof = 0)
cov_np

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [38]:
# вычисление коэффициента корреляции по формуле Пирсона
std_zp = np.std(zp)
std_ks = np.std(ks)
r_pirs = cov_el / (std_zp * std_ks)
r_pirs

0.8874900920739158

In [39]:
# вычисление коэффициента корреляции с помощью функции corrcoef
r_np = np.corrcoef(zp, ks)
r_np

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

**Ответ**: Ковариация величин zp и ks равна 9157.84, а коэффициент корреляции - 0.887


## Задание 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [40]:
n = 10
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
p = 0.95
alpha = 1 - p

In [41]:
# Вычислим математическое ожидание для выборки
M = np.mean(x)
M

118.1

In [42]:
# Вычислим несмещённое среднеквадратичное отклонение
std = np.std(x, ddof = 1)
std

10.54566788359614

In [43]:
# Используем критерий Стьюдента, поскольку среднеквадратичное отклонение неизвестно
# Из таблицы получаем значение t для alpha/2 и 9 степеней свободы:
print(alpha / 2)
t = 2.262

0.025000000000000022


In [44]:
# Стандартная ошибка среднего
SE = std / n**0.5

In [45]:
# Левая граница интервала
int_left = M - t * SE
int_left

110.55660776308164

In [46]:
# Правая граница интервала
int_right = M + t * SE
int_right

125.64339223691834

**Ответ:** Доверительный интервал для значений IQ с вероятностью 0.95: $[110,56;125,64]$

## Задание 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [47]:
M = 174.2
n = 27
D = 25
p = 0.95
alpha = 1 - p

In [48]:
# Используем z-критерий, поскольку среднеквадратичное отклонение известно
# Из таблицы получаем значение z для alpha/2:
print(alpha / 2)
z = 1.96

0.025000000000000022


In [49]:
# Стандартная ошибка среднего
SE = D**0.5 / n**0.5

In [50]:
# Левая граница интервала
int_left = M - z * SE
int_left

172.31398912064722

In [51]:
# Правая граница интервала
int_right = M + z * SE
int_right

176.08601087935276

**Ответ:** Доверительный интервал для математического ожидания с надёжностью 0.95: $[172,31;176,09]$


